### PR TITLE
Add release channel option to publish job

### LINKED ISSUE #48627: Add release tag dropdown to release job
It's annoying to remember to set a new release as "latest" after the release has completed. We should add a new optional parameter to the release publish job that lets you set a release channel for the release.

### PR DESCRIPTION
**Pull request by iethree**
closes https://github.com/metabase/metabase/issues/48627

## Description

This does a few things to simplify the release workflow

1. Removes now-irrelevant options to skip certain release steps. We only wanted to skip these steps in order to release more "slowly". Now that we have release channels, and a separate job to set them on demand, as well as patches that we can release quietly, we don't need to be able to skip release steps (some of which had implicit dependencies that made certain combinations of step skips impossible).
2. Combines the job-complete-message step for patches and regular releases. The only difference is whether they send a message to the #general slack channel
3. Doing the above greatly simplifies the ability to adds an input parameter to be able to (optionally) trigger the set-release-channel job at the end of a release.

Screen Shot 2024-10-23 at 4 43 15 PM

- ✅ [Test 1 - set nightly release channel](https://github.com/iethree/metabase/actions/runs/11497177638) (failing steps were irrelevant ones that don't work in my fork)
- ✅ [Test 2 - don't set release channel](https://github.com/iethree/metabase/actions/runs/11497358693)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,26 +11,16 @@ on:
...
-        type: boolean
-        default: false
-      skip-tag:
-        description: 'Skip tagging the release'
-        type: boolean
-        default: false
-      skip-release-notes:
-        description: 'Skip publishing release notes'
-        type: boolean
-        default: false
```
Comment (iethree) [inline .github/workflows/release.yml:33]: these options haven't been used and clutter up the job inputs. They're effectivly obsolete now that tagging release channels is separately controllable.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -640,6 +628,11 @@ jobs:
       AWS_S3_STATIC_BUCKET: ${{ vars.AWS_S3_STATIC_BUCKET }}
       AWS_REGION: ${{ vars.AWS_REGION }}
     steps:
+    - name: Finish early
+      if: ${{ inputs.auto }}
+      run: |
+        echo "skipping version info update for nightly patch"
+        exit 0
```
Comment (iethree) [inline .github/workflows/release.yml:635]: we want this job to always run, so we can make sure that the tagging job always runs after it, if at all

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -719,42 +710,15 @@ jobs:
...
+            const isAutoRelease = '${{ inputs.auto }}' === 'true';
 
             await sendPublishCompleteMessage({
               owner: context.repo.owner,
               repo: context.repo.repo,
               version: '${{ inputs.version }}',
               runId: '${{ github.run_id }}',
               channelName: '${{ vars.SLACK_RELEASE_CHANNEL }}',
-              generalChannelName: '',
+              generalChannelName: !isAutoRelease ? '${{ vars.SLACK_GENERAL_CHANNEL }}' : '',
```
Comment (iethree) [inline .github/workflows/release.yml:721]: we can replace two different release complete jobs with one with this simple ternary 😊

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -11,26 +11,16 @@ on:
...
+      release-channel:
+        description: 'The release channel to publish to (optional)'
+        type: choice
+        options:
+          - none
+          - latest
+          - beta
+          - nightly
+        required: true
+        default: 'none'
```
Comment (appleby) [inline .github/workflows/release.yml:23]: release noob question: in what situations do we set release-channel to 'none'? Just for testing workflows?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -719,42 +706,15 @@ jobs:
...
-          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
-        with:
-          script: | # js
-            const { sendPublishCompleteMessage } = require('${{ github.workspace }}/release/dist/index.cjs');
+            const isAutoRelease = '${{ inputs.auto }}' === 'true';
 
             await sendPublishCompleteMessage({
               owner: context.repo.owner,
               repo: context.repo.repo,
               version: '${{ inputs.version }}',
```
Comment (appleby) [inline .github/workflows/release.yml:714]: Is release channel info included the slack message (or useful to include)?

**Review by appleby — APPROVED** [review]
LGTM but this is my first release workflow review, so feel free to ignore.

File: .github/workflows/release.yml (hunk 4 above)
Comment (appleby) [reply .github/workflows/release.yml:23]: possibly found the answer in the other PR

https://github.com/metabase/metabase/pull/49093#discussion_r1828170049

Comment (iethree) [reply .github/workflows/release.yml:23]: some situations where we might want to use the "none" release channel

1) patches or minor releases for older versions
2) minor releases for new versions that we may want to let more slowly percolate into the wild before setting them to "stable"

File: .github/workflows/release.yml (hunk 5 above)
Comment (iethree) [reply .github/workflows/release.yml:714]: good idea! i'll submit a follow-up PR for that

**Pull request merged**